Userland/Libraries/LibTextCodec/Encoder.cpp

```diff
@@ -0,0 +1,528 @@
+            continue;
+        }
+
+        // 6. If code point is U+2212, set it to U+FF0D.
+        if (item == 0x2212)
+            item = 0xFF0D;
+
+        // 7. Let pointer be the index Shift_JIS pointer for code point.
+        auto pointer = index_shift_jis_pointer(item);
+
+        // 8. If pointer is null, return error with code point.
+        if (!pointer.has_value()) {
+            TRY(on_error(item));
+            continue;
+        }
+
+        // 9. Let lead be pointer / 188.
+        auto lead = *pointer / 188;
+
+        // 10. Let lead offset be 0x81 if lead is less than 0x1F, otherwise 0xC1.
+        auto lead_offset = 0xC1;
+        if (lead < 0x1F)
+            lead_offset = 0x81;
+
+        // 11. Let trail be pointer % 188.
+        auto trail = *pointer % 188;
+
+        // 12. Let offset be 0x40 if trail is less than 0x3F, otherwise 0x41.
+        auto offset = 0x41;
+        if (trail < 0x3F)
+            offset = 0x40;
+
+        // 13. Return two bytes whose values are lead + lead offset and trail + offset.
+        TRY(on_byte(static_cast<u8>(lead + lead_offset)));
+        TRY(on_byte(static_cast<u8>(trail + offset)));
+    }
+
+    return {};
+}
+
+// https://encoding.spec.whatwg.org/#euc-kr-encoder
+ErrorOr<void> EUCKREncoder::process(Utf8View input, Function<ErrorOr<void>(u8)> on_byte, Function<ErrorOr<void>(u32)> on_error)
+{
+    for (u32 item : input) {
+        // 1. If code point is end-of-queue, return finished.
+
+        // 2. If code point is an ASCII code point, return a byte whose value is code point.
+        if (is_ascii(item)) {
+            TRY(on_byte(static_cast<u8>(item)));
+            continue;
+        }
+
+        // 3. Let pointer be the index pointer for code point in index EUC-KR.
+        auto pointer = code_point_euc_kr_index(item);
+
+        // 4. If pointer is null, return error with code point.
+        if (!pointer.has_value()) {
+            TRY(on_error(item));
+            continue;
+        }
+
+        // 5. Let lead be pointer / 190 + 0x81.
+        auto lead = *pointer / 190 + 0x81;
+
+        // 6. Let trail be pointer % 190 + 0x41.
+        auto trail = *pointer % 190 + 0x41;
+
+        // 7. Return two bytes whose values are lead and trail.
+        TRY(on_byte(static_cast<u8>(lead)));
+        TRY(on_byte(static_cast<u8>(trail)));
+    }
+
+    return {};
+}
+
+// https://encoding.spec.whatwg.org/#big5-encoder
+ErrorOr<void> Big5Encoder::process(Utf8View input, Function<ErrorOr<void>(u8)> on_byte, Function<ErrorOr<void>(u32)> on_error)
+{
+    for (u32 item : input) {
+        // 1. If code point is end-of-queue, return finished.
+
+        // 2. If code point is an ASCII code point, return a byte whose value is code point.
+        if (is_ascii(item)) {
+            TRY(on_byte(static_cast<u8>(item)));
+            continue;
+        }
+
+        // 3. Let pointer be the index Big5 pointer for code point.
+        auto pointer = code_point_big5_index(item);
```

Comment:
I think the Big5 errors we're seeing may be because we don't implement these steps here: https://encoding.spec.whatwg.org/#index-big5-pointer.

Reply:
I've implemented `index-big5-pointer`, and those tests do now pass.